src/components/header/export-modal/renderer.tsx

```diff
@@ -77,10 +86,15 @@ class ExportModal extends React.PureComponent<Props, State> {
     ) {
       return;
     }
-    const content = this.state.downloadVegaJSON ? this.props.vegaSpec : this.props.vegaLiteSpec;
+    let content = this.state.downloadVegaJSON ? this.props.vegaSpec : this.props.vegaLiteSpec;
     const filename = this.state.downloadVegaJSON ? `visualization.vg.json` : `visualization.vl.json`;
+    if (this.state.includeConfig) {
+      content = mergeDeep(content, { config: this.props.config });
```

Comment:
Please confirm that the spec.config has precedence over the config. Also confirm that the behavior is consistent (e.g. make a spec.config and a config that set the background color and see what happens.

Reply:
Sure!

Reply:
Made the change!
Findings
1. Even if the config is written , the vega automatically takes the spec.config as of higher priority.
2. The fix includes checking if the config already exists ? dont merge : merge

Reply:
This is incorrect. We specifically want to merge even if the config exists in the spec already.

Reply:
Should we merge spec.config and config giving spec.config a priority?

Reply:
Yes (if that's consistent with Vega's behavior).